### PR TITLE
Feature/rst 529 list of users appliations

### PR DESCRIPTION
**Pull request by zaparka**
This is solving the issue for users when they forget to write down the reference number of the application that they processed. It also exposes 'open' applications that user 'abandoned'.

### CODE REVIEW

File: app/helpers/home_helper.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  private
+
+  def waiting_for_evidence_path(application)
+    record = Views::ApplicationList.new(application.evidence_check)
+    evidence_path(record.evidence_or_part_payment)
+  end
+
+  def waiting_for_part_payment(application)
```
Comment (leoapost) [inline app/helpers/home_helper.rb]: this method has the same code with `waiting_for_evidence_path`. Can we dry it out?

File: app/models/query/last_updated_applications.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Query
+  class LastUpdatedApplications
+    def initialize(user)
+      @user = user
+    end
+
+    def find
+      @user.applications.order(updated_at: :desc).limit(5)
```
Comment (leoapost) [inline app/models/query/last_updated_applications.rb]: This `.limit()` may cause confusion when using this class. 

If it is necessary then how about something like this:
```
def find(limit: nil)
  @user.applications.order(updated_at: :desc).limit(limit)
end
```
and then 
- `Query::LastUpdatedApplications.new.find` will return all the last updated applications
- `Query::LastUpdatedApplications.new.find(limit: 10)` will return the last 10 updated applications

File: app/helpers/home_helper.rb (hunk 1 above)
Comment (zaparka) [reply app/helpers/home_helper.rb]: well spotted.

File: app/models/query/last_updated_applications.rb (hunk 2 above)
Comment (zaparka) [reply app/models/query/last_updated_applications.rb]: yeah make sense

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -48,6 +49,10 @@ def load_waiting_applications
     end
   end
 
+  def load_users_last_applications
+    @last_updated_applications = Query::LastUpdatedApplications.new(current_user).find(limit: 5)
```
Comment (leoapost) [inline app/controllers/home_controller.rb]: I forgot to mention we may also need to memoize the `@last_updated_applications`.

Comment (zaparka) [reply app/controllers/home_controller.rb]: sure

**Pull request merged**